src/commands/sarif/upload.ts

```diff
@@ -0,0 +1,199 @@
+import fs from 'fs'
+import path from 'path'
+
+import Ajv from 'ajv'
+import addFormats from 'ajv-formats'
+import chalk from 'chalk'
+import {Command} from 'clipanion'
+import glob from 'glob'
+import asyncPool from 'tiny-async-pool'
+
+import {getCISpanTags} from '../../helpers/ci'
+import {getGitMetadata} from '../../helpers/git/format-git-span-data'
+import {SpanTags} from '../../helpers/interfaces'
+import {retryRequest} from '../../helpers/retry'
+import {parseTags} from '../../helpers/tags'
+import {getUserGitSpanTags} from '../../helpers/user-provided-git'
+import {buildPath} from '../../helpers/utils'
+
+import {apiConstructor} from './api'
+import {APIHelper, Payload} from './interfaces'
+import sarifJsonSchema from './json-schema/sarif-schema-2.1.0.json'
+import {
+  renderCommandInfo,
+  renderSuccessfulCommand,
+  renderDryRunUpload,
+  renderRetriedUpload,
+  renderFailedUpload,
+  renderInvalidFile,
+} from './renderer'
+import {getBaseIntakeUrl} from './utils'
+
+const errorCodesStopUpload = [400, 403]
+
+const ajv = new Ajv()
+addFormats(ajv)
+
+const validateSarif = (sarifReportPath: string) => {
+  const sarifJsonSchemaValidate = ajv.compile(sarifJsonSchema)
+  try {
+    const sarifReportContent = JSON.parse(String(fs.readFileSync(sarifReportPath)))
```

Comment:
I think this should work. Let me know if it doesn't:
```suggestion
    const sarifReportContent = require(sarifReportPath)
```

Reply:
After the change, the tests are failing :(, so I guess it's not working

Reply:
yeah nvm, I thought `sarifReportPath` had a `.json` extension, but it's `.sarif`. The format should be json compatible but `require` won't work.